modules/apps/forms-and-workflow/dynamic-data-mapping/dynamic-data-mapping-type-grid/src/test/testJS/grid_field.js

```diff
@@ -0,0 +1,266 @@
+	'Liferay.DDM.Field.Grid',
+	function() {
+		var gridField;
+
+		afterEach(
+			function(done) {
+				gridField.destroy();
+
+				done();
+			}
+		);
+
+		before(
+			function(done) {
+				AUI().use(
+					'liferay-ddm-form-field-grid-template',
+					'liferay-ddm-form-field-grid',
+					function(A) {
+						Liferay.DDM.Renderer.FieldTypes.register(
+							{
+								'javaScriptClass': 'Liferay.DDM.Field.Grid',
+								'name': 'grid',
+								'templateNamespace': 'ddm.grid'
+							}
+						);
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.showErrorMessage()',
+			function() {
+				it(
+					'should exists an error container if the grid field has an error message',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.set('errorMessage', 'error');
+
+						gridField.showErrorMessage();
+
+						var container = gridField.get('container');
+
+						assert.isNotNull(container.one('.help-block'), 'The selectField has an error');
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.setValue()',
+			function() {
+				it(
+					'should return empty value if set an empty object value',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({});
+
+						assert.equal(gridField.getValue().A, undefined);
+
+						assert.equal(gridField.getValue().B, undefined);
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values if set an object with properties',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1', B: '2'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, '2');
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values if set an only option',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, undefined);
+
+						done();
+					}
+				);
+
+				it(
+					'should return object values only if exists in the UI',
+					function(done) {
+						gridField = createGridField();
+
+						gridField.setValue({A: '1', B: '2', C: '3'});
+
+						assert.isObject(gridField.getValue());
+
+						assert.equal(gridField.getValue().A, '1');
+
+						assert.equal(gridField.getValue().B, '2');
+
+						assert.equal(gridField.getValue().C, undefined);
+
+						assert.isObject(gridField.get('value'));
+
+						assert.equal(gridField.get('value').A, '1');
+
+						assert.equal(gridField.get('value').B, '2');
+
+						assert.equal(gridField.get('value').C, undefined);
+
+						done();
+					}
+				);
+			}
+		);
+
+		describe(
+			'.getValue()',
+			function() {
+				it(
+					'should return object value if grid has a value in context',
+					function(done) {
+						gridField = createGridField(
+							{
```

Comment:
To make sure the value came from the context you should create the Grid using the attribute `context`
`createGridField(
  {
    context: {
      value: {B: '2'}
    }
  }
)`